queue/abstract/driver/fifo.lua

```diff
@@ -6,35 +6,54 @@ local str_type = require('queue.compat').str_type
 local tube = {}
 local method = {}
 
+-- validate space of queue
+local function validate_space(space)
+    -- check indexes
+    local indexes = {'task_id', 'status'}
+    for _, index in pairs(indexes) do
+        if space.index[index] == nil then
+            error(string.format('space "%s" does not have "%s" index',
+                space.name, index))
+        end
+    end
+end
+
 -- create space
 function tube.create_space(space_name, opts)
     local space_opts         = {}
     local if_not_exists      = opts.if_not_exists or false
     space_opts.temporary     = opts.temporary or false
-    space_opts.if_not_exists = if_not_exists
     space_opts.engine        = opts.engine or 'memtx'
     space_opts.format = {
         {name = 'task_id', type = num_type()},
         {name = 'status', type = str_type()},
         {name = 'data', type = '*'}
     }
 
-    local space = box.schema.create_space(space_name, space_opts)
-    space:create_index('task_id', {
-        type = 'tree',
-        parts = {1, num_type()},
-        if_not_exists = if_not_exists
-    })
-    space:create_index('status',  {
-        type = 'tree',
-        parts = {2, str_type(), 1, num_type()},
-        if_not_exists = if_not_exists
-    })
+    local space = box.space[space_name]
+    if if_not_exists and space then
+        -- Validate the existing space.
+        validate_space(box.space[space_name])
```

Comment:
Raw idea: should not we verify that `box.space._queue` contains the relevant entry when the requested space exists and `if_not_exists` is positive? (We can do it for all drivers in `create_tube()` in abstract.lua.)

Reply:
No. Now a tube space validation and the "_queue" space are located at different levels of abstraction (driver vs abstract).

Reply:
Okay.